AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c

```diff
@@ -88,8 +88,46 @@ int RiotStart(uint8_t *CDI, uint16_t CDILen, const char *RegistrationId)
     uint32_t            length;
     uint32_t            dcType;
 
-    // Update subject common of alias certificate TBD data to input registration Id
-    x509AliasTBSData.SubjectCommon = RegistrationId;
+    // Verify registrationId input and given an empty input, generate one based on MAC address and firmware version of your DevKit
+    int regIdLength = strlen(RegistrationId);
+    if (regIdLength == 0){
+        char * registrationId = (char*)malloc(AUTO_GEN_REGISTRATION_ID_MAX_LENGTH);
+        memset(registrationId, 0, AUTO_GEN_REGISTRATION_ID_MAX_LENGTH);
+        char * macAddress = GetBoardID();
+        char * fmVersion = getDevkitVersion();
+        LogInfo("DevKit Firmware Version: %s", fmVersion);
```

Comment:
You can LogInfo("%s  %s", fmVersion, macAddress); if you want to log MAC address

Reply:
here macAddress is not real MAC address but in the format of az-macaddress, so I get real MAC address from SystemVariables.cpp.